### PR TITLE
Flatten the RFC directories for the RFC process

### PR DESCRIPTION
**Pull request by kylef**
This pull request removes the directory hierarchies from the RFC process to simplify it, and also make it easier to determine the next RFC number. This is also a first step in aligning the RFC process for API Blueprint with the [Refract RFC Process](http://github.com/refractproject/rfcs).

Can @smizell and @zdne please review these changes. And also let me know if there are any other changes you would like to see to the RFC process here.

### CODE REVIEW

File: rfcs/0005-authentication-oauth2.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ---
-RFC: XXXX
+RFC: 5
```
Comment (kylef) [inline rfcs/0005-authentication-oauth2.md:2]: This draft wasn't defined an RFC number before, so assigned it here.

File: rfcs/0002-authentication.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ---
-RFC: 0002
+RFC: 2
```
Comment (kylef) [inline rfcs/0002-authentication.md:2]: I've changed the RFC numbers to not include the preceding 000's, it's easier to read without them.

Comment (smizell) [inline rfcs/0002-authentication.md:2]: I like this. I think we should keep the `0002` in the filenames like you have it for sorting purposes, but this does help in the actual document where it's not needed.

**Comment by smizell** [comment]
This looks good to me!

File: rfcs/0002-authentication.md (hunk 2 above)
Comment (zdne) [inline rfcs/0002-authentication.md:2]: :+1: 

@kylef I am up for merging this but I would stick with the leading zeroes as @smizell suggests

**Comment by zdne** [comment]
Good to merge after  https://github.com/apiaryio/api-blueprint-rfcs/pull/13#discussion_r50827734

File: rfcs/0002-authentication.md (hunk 2 above)
Comment (kylef) [inline rfcs/0002-authentication.md:2]: @zdne Stephen stated how it is working in this PR. Do you mean something different?

Comment (zdne) [inline rfcs/0002-authentication.md:2]: @kylef OK sorry. I have misread it.

**Pull request merged**